src/core/server/plugins/plugin_context.ts

```diff
@@ -122,6 +122,7 @@ export function createPluginSetupContext<TPlugin, TPluginDependencies>(
       registerOnPreAuth: deps.http.registerOnPreAuth,
       registerAuth: deps.http.registerAuth,
       registerOnPostAuth: deps.http.registerOnPostAuth,
+      registerRouter: deps.http.registerRouter,
```

Comment:
probably we can wait for documentation to be ready before exposing this API to the plugins